corim/signer.go

```diff
@@ -127,31 +127,47 @@ func getAlgAndKeyFromJWK(j []byte) (cose.Algorithm, crypto.Signer, error) {
 		return noAlg, nil, err
 	}
 
-	var key crypto.Signer
+	var key any
 
-	err = k.Raw(&key)
+	err = jwk.Export(k, &key)
 	if err != nil {
 		return noAlg, nil, err
 	}
 
-	switch v := key.(type) {
-	case *ecdsa.PrivateKey:
-		alg = ellipticCurveToAlg(v.Curve)
+	fromCurve := func(c elliptic.Curve) (cose.Algorithm, any, error) {
+		alg = ellipticCurveToAlg(c)
 		if alg == noAlg {
 			return noAlg, nil, fmt.Errorf("unknown elliptic curve %v", crv)
 		}
-	case ed25519.PrivateKey:
-		alg = cose.AlgorithmEd25519
-	case *rsa.PrivateKey:
+		return alg, key, nil
+	}
+	isRsa := func() (cose.Algorithm, any, error) {
 		alg = rsaJWKToAlg(k)
 		if alg == noAlg {
-			return noAlg, nil, fmt.Errorf("unknown RSA algorithm %q", k.Algorithm().String())
+			name := "unnamed"
+			if jalg, ok := k.Algorithm(); ok {
+				name = jalg.String()
+			}
+			return noAlg, nil, fmt.Errorf("unknown RSA algorithm %q", name)
 		}
+		return alg, key, nil
+	}
+	switch v := key.(type) {
+	case *ecdsa.PrivateKey:
```

Comment:
The linter bizarrely crashes around here with the error:
```
Error: corim/signer.go:156:7: previous case (typecheck)
	case *ecdsa.PrivateKey:
	     ^
```

Reply:
🤷

Reply:
I suggest we use `staticcheck` instead.

Reply:
I've created a PR to fix all the linter problems except the documentation lint that I can rebase on top of when it's merged.